var/spack/packages/hdf5/package.py

```diff
@@ -10,12 +10,15 @@ class Hdf5(Package):
     url      = "http://www.hdfgroup.org/ftp/HDF5/releases/hdf5-1.8.13/src/hdf5-1.8.13.tar.gz"
     list_url = "http://www.hdfgroup.org/ftp/HDF5/releases"
     list_depth = 3
-    
+
     version('1.8.16', 'b8ed9a36ae142317f88b0c7ef4b9c618')
     version('1.8.15', '03cccb5b33dbe975fdcd8ae9dc021f24')
     version('1.8.13', 'c03426e9e77d7766944654280b467289')
 
+    variant('cxx', default=False, description='Enable C++ support')
+    variant('fortran', default=False, description='Enable Fortran support')
```

Comment:
FYI…

I think c++ and fortran interfaces to HDF5 are less well used/tested than C interface. But, thats just my opinion and not based on any real usage information.

Nonetheless, I opted to default these to off to minimize likelihood of failure to build HDF5 as other packages I am aware of that require it don't need 'em.

But, these opinions are just based on being cautious and not ever really having needed c++ or fortran interfaces to HDF5.

Mark




From: Todd Gamblin <notifications@github.com<mailto:notifications@github.com>>
Reply-To: LLNL/spack <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Thursday, December 31, 2015 3:29 PM
To: LLNL/spack <spack@noreply.github.com<mailto:spack@noreply.github.com>>
Cc: "Miller, Mark C." <miller86@llnl.gov<mailto:miller86@llnl.gov>>
Subject: Re: [spack] Enable several variants for HDF5 (#278)


In var/spack/packages/hdf5/package.py<https://github.com/LLNL/spack/pull/278#discussion_r48673904>:

>      version('1.8.16', 'b8ed9a36ae142317f88b0c7ef4b9c618')
>      version('1.8.15', '03cccb5b33dbe975fdcd8ae9dc021f24')
>      version('1.8.13', 'c03426e9e77d7766944654280b467289')
>
> +    variant('cxx', default=False, description='Enable C++ support')
> +    variant('fortran', default=False, description='Enable Fortran support')


Can you make these two enabled by default? I think in the common case we might as well build the C++ and Fortran interfaces, unless there's a good reason not to. @markcmiller86<https://github.com/markcmiller86> -- opinions?

—
Reply to this email directly or view it on GitHub<https://github.com/LLNL/spack/pull/278/files#r48673904>.